src/Functional.CQS.AOP.IoC.SimpleInjector.MetricsCapturing/ConventionBasedDecoratorRegistrationGatewayExtensions.cs

```diff
@@ -0,0 +1,101 @@
+using Functional.CQS.AOP.IoC.SimpleInjector.MetricsCapturing;
+using Functional.CQS.AOP.IoC.SimpleInjector.MetricsCapturing.Configuration;
+using Functional.CQS.AOP.IoC.SimpleInjector.Models;
+using Functional.CQS.AOP.MetricsCapturing;
+
+// ReSharper disable once CheckNamespace
+namespace SimpleInjector
+{
+	/// <summary>
+	/// Extension methods for <see cref="ConventionBasedDecoratorRegistrationGateway"/>.
+	/// </summary>
+	public static class ConventionBasedDecoratorRegistrationGatewayExtensions
+	{
+		/// <summary>
+		/// Register all core components required for applying metrics-capturing decorators to Functional.CQS handler implementations.
+		/// Only handlers with corresponding <see cref="IMetricsCapturingStrategyForQuery{TQuery, TResult}"/> and <see cref="IMetricsCapturingStrategyForCommand{TCommand, TError}"/> implementations will have the handler-specific metrics-capturing decorator applied to them.
+		/// No universal metrics-capturing decorator will be applied.
+		/// </summary>
+		/// <param name="gateway">The gateway.</param>
+		/// <param name="configurationParameters">The configuration parameters.</param>
+		/// <returns></returns>
+		public static ConventionBasedDecoratorRegistrationGateway WithMetricsCapturingDecorator(this ConventionBasedDecoratorRegistrationGateway gateway, MetricsCapturingModuleConfigurationParameters configurationParameters)
+		{
+			gateway.RegisterMetricsCapturingDecoratorForIndividualQueryHandlerImplementations(configurationParameters);
+			gateway.RegisterMetricsCapturingDecoratorForIndividualCommandHandlerImplementations(configurationParameters);
+
+			return gateway;
+		}
+
+		/// <summary>
+		/// Register all core components required for applying metrics-capturing decorators to Functional.CQS handler implementations.
+		/// Only handlers with corresponding <see cref="IMetricsCapturingStrategyForQuery{TQuery, TResult}"/> and <see cref="IMetricsCapturingStrategyForCommand{TCommand, TError}"/> implementations will have the handler-specific metrics-capturing decorator applied to them.
+		/// All handler implementations will have the universal metrics-capturing decorator applied to them.
+		/// </summary>
+		/// <typeparam name="TUniversalMetricsCapturingStrategy">The universal metrics capturing strategy type.</typeparam>
+		/// <param name="gateway">The gateway.</param>
+		/// <param name="configurationParameters">The configuration parameters.</param>
+		/// <returns></returns>
+		public static ConventionBasedDecoratorRegistrationGateway WithMetricsCapturingDecorator<TUniversalMetricsCapturingStrategy>(this ConventionBasedDecoratorRegistrationGateway gateway, MetricsCapturingModuleConfigurationParameters configurationParameters)
+			where TUniversalMetricsCapturingStrategy : class, IUniversalMetricsCapturingStrategy
+		{
+			gateway.RegisterMetricsCapturingDecoratorForIndividualQueryHandlerImplementations(configurationParameters);
```

Comment:
Rather than writing the following pair of statements in two separate methods:
```
gateway.RegisterMetricsCapturingDecoratorForIndividualQueryHandlerImplementations(configurationParameters);
gateway.RegisterMetricsCapturingDecoratorForIndividualCommandHandlerImplementations(configurationParameters);
```

would it make sense to replace the duplicate lines in this method with this call:
```
gateway
	.WithMetricsCapturingDecorator(configurationParameters)
	.RegisterUniversalMetricsCapturingDecorator<TUniversalMetricsCapturingStrategy>(configurationParameters);

return gateway;
```

You could also use:
```
gateway.WithMetricsCapturingDecorator(configurationParameters)
gateway.RegisterUniversalMetricsCapturingDecorator<TUniversalMetricsCapturingStrategy>(configurationParameters);

return gateway;
```

Reply:
Excellent idea!